R/validation_rules.R

```diff
@@ -0,0 +1,70 @@
+#' Validate page size
+#'
+#' @param page_size Chosen page size
+#' @param min Minimum valid page_size for EES API
+#' @param max Maximum valid page_size for EES API
+#'
+#' @return Logic
+#' @export
+#'
+#' @examples
+#' validate_page_size(20)
+validate_page_size <- function(page_size, min = 1, max = 40) {
+  if (!is.null(page_size)) {
+    if (is.numeric(page_size)) {
+      valid <- dplyr::between(page_size, 1, 40)
+    } else {
+      valid <- FALSE
+    }
+    if (!valid) {
+      stop(
+        "The page size can only be a numeric value within the range 1 <= page_size <= 40."
+      )
+    }
+  }
+}
+
+#' Validate element IDs
+#'
+#' @param element_id ID for publication or a data set
+#' @param level ID level: "publication" or "dataset"
+#'
+#' @return NULL
+#' @export
+#'
+#' @examples
+#' validate_ees_id(example_id("publication"), level = "publication")
+validate_ees_id <- function(element_id, level = "publication") {
+  if (!(level %in% c("publication", "dataset"))) {
+    stop(
+      paste0(
+        "Non-valid element level received by validate_id.\n",
+        'Should be one of "publication" or "dataset".'
+      )
+    )
+  }
+  if (is.null(element_id) || is.na(element_id)) {
+    stop(
+      "The variable ", level,
+      "_id is NULL or NA, please provide a valid ", level,
+      "_id."
+    )
+  } else if (level %in% c("publication", "dataset")) {
```

Comment:
Feel like you don't need this last else if and the err_string definition makes more sense outside of your if/elses?

Reply:
As it stands yes, but I partially wrote that bit with the idea that we might add other ids into this validation that would then have a different structure / number of characters to dataset_id and publication_id. Basically I'd been thinking of adding to this one when I write things that involve filter and indicator ids.